src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/DictionaryExtenstions.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+
+    internal static class DictionaryExtenstions
+    {
+        internal class AddResult
+        {
+            public readonly bool KeyChanged;
+            public readonly string OldKey;
+            public readonly string NewKey;
+
+            public AddResult()
+            {
+                KeyChanged = false;
+            }
+
+            public AddResult(string oldKey, string newKey)
+            {
+                this.KeyChanged = true;
+                this.OldKey = oldKey;
+                this.NewKey = newKey;
+            }
+        }
+
+        public static AddResult AddOrDuplicate(this IDictionary<string, object> dictionary, KeyValuePair<string, object> pair)
+        {
+            return AddOrDuplicate(dictionary, pair.Key, pair.Value);
+        }
+
+        public static AddResult AddOrDuplicate(this IDictionary<string, object> dictionary,  string key,  object value)
+        {
+            Validation.Requires.NotNull(dictionary, nameof(dictionary));
+            Validation.Requires.NotNull(key, nameof(key));
+
+            if (!dictionary.ContainsKey(key))
+            {
+                dictionary.Add(key, value);
+                return new AddResult();
+            }
+
+            Random random = new Random(DateTime.Now.Millisecond);
```

Comment:
1. No need to use Random(seed) constructor; the parameter-less constructor does what your code does
2. How about creating a static instance of the Random class and caching it?